lib/app/services/storage.dart

```diff
@@ -1,50 +1,56 @@
+import 'dart:convert';
+
 import 'package:get_it/get_it.dart';
 import 'package:meta/meta.dart';
-import 'package:schulcloud/assignment/assignment.dart';
-import 'package:schulcloud/course/course.dart';
-import 'package:schulcloud/file/file.dart';
-import 'package:schulcloud/news/news.dart';
 import 'package:streaming_shared_preferences/streaming_shared_preferences.dart';
 
 import '../app.dart';
 import '../hive.dart';
 
 /// A service that offers storage of app-wide data.
 class StorageService {
-  StorageService._(
-    this._prefs,
-    this.userIdString,
-    this.email,
-    this.token,
-    this.cache,
-  );
+  StorageService._({
+    StreamingSharedPreferences prefs,
+    @required this.userIdString,
+    @required this.email,
+    @required this.token,
+    @required this.root,
+  }) : _prefs = prefs;
+
+  // The token is a JWT token. JWT tokens consist of a header, body and claim
+  // (signature), all separated by dots and encoded in base64. For now, we
+  // don't verify the claim, but just decode the body.
+  Id<User> get currentUserId {
+    final token = this.token.getValue();
+    final encodedBody = token.split('.')[1];
+    final body = String.fromCharCodes(base64.decode(encodedBody));
+    return Id<User>(json.decode(body)['userId']);
+  }
```

Comment:
We don't need to decode the token every time as we already store `userIdString`.
```suggestion
  Id<User> get userId => Id<User>(userIdString.getValue());
```